core/commands/repo.go

```diff
@@ -87,26 +110,29 @@ order to reclaim hard disk space.
 				return nil, err
 			}
 
-			marshal := func(v interface{}) (io.Reader, error) {
-				obj, ok := v.(*corerepo.KeyRemoved)
+			for v := range outChan {
```

Comment:
I am not 100% sure I am following, but I attempted to do what you want.  It did not seam to help with the duplicate error reporting.